Patch-Source: https://github.com/emersion/xdg-desktop-portal-wlr/pull/353
---
From 4737535282f2d74c15ea0357d48a2ecf2154a623 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Mon, 17 Nov 2025 13:37:51 +0100
Subject: [PATCH] build: add default for systemd user_unit_dir

Allows building the service files without a systemd dependency by
defining a default systemd user unit directory.

Relevant in Alpine, where we can package the service files in a
subpackage without having systemd in Alpine.
---
 meson.build | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 77e08bc6..2beefd64 100644
--- a/meson.build
+++ b/meson.build
@@ -89,11 +89,16 @@ conf_data = configuration_data()
 conf_data.set('libexecdir', get_option('prefix') / get_option('libexecdir'))
 conf_data.set('systemd_service', '')
 
-systemd = dependency('systemd', required: get_option('systemd'))
-if systemd.found()
+systemd = dependency('systemd', required: false)
+if (systemd.found() and get_option('systemd').allowed()) or get_option('systemd').enabled()
+	if systemd.found()
+		user_unit_dir = systemd.get_variable(pkgconfig: 'systemduserunitdir',
+			pkgconfig_define: ['prefix', get_option('prefix')])
+	else
+		user_unit_dir = get_option('prefix')/ 'lib' / 'systemd' / 'user'
+	endif
+
 	systemd_service_file = 'xdg-desktop-portal-wlr.service'
-	user_unit_dir = systemd.get_variable(pkgconfig: 'systemduserunitdir',
-		pkgconfig_define: ['prefix', get_option('prefix')])
 	conf_data.set('systemd_service', 'SystemdService=' + systemd_service_file)
 
 	configure_file(
